Index: plugin.xml
===================================================================
--- plugin.xml	(revision 12229)
+++ plugin.xml	(working copy)
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <?eclipse version="3.0"?>
 <plugin>
+   <extension-point id="filters" name="URML Diagram Filters" schema="schema/org.unicase.ui.diagram.urmlDiagram.filters.exsd"/>
 
    <extension point="org.eclipse.team.core.fileTypes" id="repo-ftype">
       <?gmfgen generated="true"?>
Index: src/org/unicase/ui/diagram/urml/edit/parts/URMLDiagramEditPart.java
===================================================================
--- src/org/unicase/ui/diagram/urml/edit/parts/URMLDiagramEditPart.java	(revision 12229)
+++ src/org/unicase/ui/diagram/urml/edit/parts/URMLDiagramEditPart.java	(working copy)
@@ -1,16 +1,26 @@
 package org.unicase.ui.diagram.urml.edit.parts;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.Platform;
 import org.eclipse.gmf.runtime.diagram.ui.editpolicies.EditPolicyRoles;
 import org.eclipse.gmf.runtime.notation.View;
 import org.unicase.ui.diagram.urml.edit.policies.URMLDiagramCanonicalEditPolicy;
 import org.unicase.ui.diagram.urml.edit.policies.URMLDiagramItemSemanticEditPolicy;
 import org.unicase.ui.unicasecommon.diagram.edit.parts.MEDiagramEditPart;
+import org.unicase.ui.urml.UrmlDiagramFilter;
+import org.unicase.workspace.util.WorkspaceUtil;
 
 /**
  * @generated
  */
 public class URMLDiagramEditPart extends MEDiagramEditPart {
 
+	
 	/**
 	 * @generated
 	 */
@@ -20,6 +30,8 @@
 	 * @generated
 	 */
 	public static final int VISUAL_ID = 1000;
+	
+	private static final String EXTENSION_POINT_ID = "org.unicase.ui.diagram.urmlDiagram.filters";
 
 	/**
 	 * @generated
@@ -37,5 +49,30 @@
 		installEditPolicy(EditPolicyRoles.CANONICAL_ROLE, new URMLDiagramCanonicalEditPolicy());
 		// removeEditPolicy(org.eclipse.gmf.runtime.diagram.ui.editpolicies.EditPolicyRoles.POPUPBAR_ROLE);
 	}
-
+	
+	@Override
+	protected List<?> getModelChildren() {
+		List<?> filterElements = super.getModelChildren();
+		
+		IConfigurationElement[] filterExtensions = Platform.getExtensionRegistry().getConfigurationElementsFor(
+				EXTENSION_POINT_ID);
+			for (IConfigurationElement e : filterExtensions) {
+				String className = e.getAttribute("filterClass");
+				try{
+					Class<?> resolvedClass = Class.forName(className);
+					UrmlDiagramFilter filter = (UrmlDiagramFilter) resolvedClass.newInstance();
+					filterElements = filter.filter(filterElements);
+				} catch (ClassNotFoundException e1) {
+					WorkspaceUtil.logException("", e1);
+				} catch (InstantiationException e3) {
+					// TODO Auto-generated catch block
+					e3.printStackTrace();
+				} catch (IllegalAccessException e2) {
+					// TODO Auto-generated catch block
+					e2.printStackTrace();
+				}
+			}
+		return filterElements;
+	}
 }
+
Index: src/org/unicase/ui/urml/UrmlDiagramFilter.java
===================================================================
--- src/org/unicase/ui/urml/UrmlDiagramFilter.java	(revision 0)
+++ src/org/unicase/ui/urml/UrmlDiagramFilter.java	(revision 0)
@@ -0,0 +1,9 @@
+package org.unicase.ui.urml;
+
+import java.util.List;
+
+public interface UrmlDiagramFilter {
+
+	List<?> filter(List<?> elements);
+
+}
Index: schema/org.unicase.ui.diagram.urmlDiagram.filters.exsd
===================================================================
--- schema/org.unicase.ui.diagram.urmlDiagram.filters.exsd	(revision 0)
+++ schema/org.unicase.ui.diagram.urmlDiagram.filters.exsd	(revision 0)
@@ -0,0 +1,102 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!-- Schema file written by PDE -->
+<schema targetNamespace="org.unicase.ui.diagram.urmlDiagram" xmlns="http://www.w3.org/2001/XMLSchema">
+<annotation>
+      <appInfo>
+         <meta.schema plugin="org.unicase.ui.diagram.urmlDiagram" id="org.unicase.ui.diagram.urmlDiagram.filters" name="URML Diagram Filters"/>
+      </appInfo>
+      <documentation>
+         [Enter description of this extension point.]
+      </documentation>
+   </annotation>
+
+   <element name="extension">
+      <annotation>
+         <appInfo>
+            <meta.element />
+         </appInfo>
+      </annotation>
+      <complexType>
+         <sequence>
+            <element ref="filter"/>
+         </sequence>
+         <attribute name="point" type="string" use="required">
+            <annotation>
+               <documentation>
+                  
+               </documentation>
+            </annotation>
+         </attribute>
+         <attribute name="id" type="string">
+            <annotation>
+               <documentation>
+                  
+               </documentation>
+            </annotation>
+         </attribute>
+         <attribute name="name" type="string">
+            <annotation>
+               <documentation>
+                  
+               </documentation>
+               <appInfo>
+                  <meta.attribute translatable="true"/>
+               </appInfo>
+            </annotation>
+         </attribute>
+      </complexType>
+   </element>
+
+   <element name="filter">
+      <complexType>
+         <attribute name="filterClass" type="string" use="required">
+            <annotation>
+               <documentation>
+                  
+               </documentation>
+               <appInfo>
+                  <meta.attribute kind="java" basedOn=":org.unicase.ui.urml.UrmlDiagramFilter"/>
+               </appInfo>
+            </annotation>
+         </attribute>
+      </complexType>
+   </element>
+
+   <annotation>
+      <appInfo>
+         <meta.section type="since"/>
+      </appInfo>
+      <documentation>
+         [Enter the first release in which this extension point appears.]
+      </documentation>
+   </annotation>
+
+   <annotation>
+      <appInfo>
+         <meta.section type="examples"/>
+      </appInfo>
+      <documentation>
+         [Enter extension point usage example here.]
+      </documentation>
+   </annotation>
+
+   <annotation>
+      <appInfo>
+         <meta.section type="apiinfo"/>
+      </appInfo>
+      <documentation>
+         [Enter API information here.]
+      </documentation>
+   </annotation>
+
+   <annotation>
+      <appInfo>
+         <meta.section type="implementation"/>
+      </appInfo>
+      <documentation>
+         [Enter information about supplied implementation of this extension point.]
+      </documentation>
+   </annotation>
+
+
+</schema>
Index: META-INF/MANIFEST.MF
===================================================================
--- META-INF/MANIFEST.MF	(revision 12229)
+++ META-INF/MANIFEST.MF	(working copy)
@@ -8,7 +8,8 @@
 Bundle-Vendor: %providerName
 Export-Package: org.unicase.ui.diagram.urml.edit.parts,
  org.unicase.ui.diagram.urml.part,
- org.unicase.ui.diagram.urml.providers
+ org.unicase.ui.diagram.urml.providers,
+ org.unicase.ui.urml
 Require-Bundle: org.eclipse.core.runtime,
  org.eclipse.core.resources,
  org.eclipse.core.expressions,
